### DISCUSSION TITLE
RN new architecture in Jul 2023

### DISCUSSION THREAD

**Discussion by SAMY-MOHSEN-111** [expo/expo#23449] [Archived: Questions]
hi, I was wondering if expo now at  Jul 2023 support the new architecture? I have looked here and there but I am not 100% percent sure about it.

**Comment by Gloriouspete** [comment]
hi, this is totally unrelated to your question, but i want to ask if you uninstalled the global expo-cli to use npx or you use both together

**Comment by tsapeta** [comment]
Hey, so the current state of things with the New Architecture is as follows:
- All Expo SDK modules work just out of the box (as everything written with [Expo Modules API](https://docs.expo.dev/modules))
- There is no support for Expo Go (and I doubt we'll do that soon, the app is too complex)
- Development builds support the New Architecture as of SDK 49, you can enable it with `newArchitecture` property (read more on [BuildProperties docs](https://docs.expo.dev/versions/latest/sdk/build-properties/))

**Reply by mirik999** [reply]
I have an app wich runs on sdk 49 ( expo dev build ) , can i turn on newArchitecture and not be afraid that something will not work with new architecture ? and why it is not enabled by default ?

**Reply by tsapeta** [reply]
Yes, you can turn it on and try it out. The reason why it's not enabled by default in Expo is that it's still not enabled by default in React Native itself and not all libraries work as expected (you need to check them out individually).

**Reply by ms007** [reply]
Hi thanks for the answers. Are there any plans to update Expo Go as well?

**Reply by tsapeta** [reply]
In the future yes, but it's complicated to do right now as there is no way to include both, the old and new architectures in the same binary. So basically, it couldn't let you switch between them. When the new architecture is stable and ready, we'll just switch Expo Go to use only new one.

**Reply by TowhidKashem** [reply]
Thank you @tsapeta , just to clarify, when you say not all libraries work with the new arch you mean third party libraries right? The way I read your original message it sounds like all Expo packages work with the new arch enabled..

Also, the build property name mentioned at that link seems to be `newArchEnabled` not `newArchitecture`, was that just a typo or this is some undocumented additional setting?

**Reply by tsapeta** [reply]
Yes, I was talking about third party libraries. There is only one Expo module that doesn't support the new architecture – `expo-updates`, but we're planning to change that soon.

You're right, it is the `newArchEnabled` property.

**Reply by mattiamanzati** [reply]
> Yes, I was talking about third party libraries. There is only one Expo module that doesn't support the new architecture – `expo-updates`, but we're planning to change that soon.

Is this still true with SDK 50 as well or expo-updates got support for new architecture?

**Reply by tsapeta** [reply]
`expo-updates` still doesn't support the new architecture, but it's our priority for SDK 51.
